### DISCUSSION TITLE
Make apk for pre-release

### DISCUSSION THREAD

**Discussion by bydariogamer** [beat-feet/beat-feet#57] [General]
Please, could you build the apk of the pre-release and load it in 'Releases'? Having the source code is cool, but I don't work with Android apps (and I had never used Kotlin before to be honest). Other users might have a similar problem...
It would be cool to have the 0.2.0 now... I've looked at F-droid for the new version but it is still not published.

**Comment by pserwylo** [comment]
Great idea. I have been monitoring [the F-Droid build runs](https://f-droid.org/wiki/index.php?title=Special:RecentChanges&days=7&from=&hidebots=0&hideanons=1&hideliu=1&limit=500) regularly for the for the past month or so, and it seems perhaps there is too big a queue of apps to build (each one takes about 30mins, and there is obviously an ever increasing number of apps, thanks in part to the success of the F-Droid data team). 

I think pre-releases are a great idea, and I'll try to do that this week some time. I've been avoiding it because I don't want to have to manage signing keys (even for pre-release apps - I love that F-Droid does this for us - even to the point where I upload the F-Droid signed version to GPlay for my other apps). However in this case I think it is waranted because I'm super keen to get iterative feedback from the community.

Let me have a stab over the comming days. Best case I could setup a nightly build F-Droid repo just for this game, but if that is a bit too difficult, then I'll just revert to manually building and signing locally, then uploading to GitHub releases.

**Reply by pserwylo** [reply]
A work in progress, not quite an APK can download and install yet, but closer. Will report back here with progress. 

https://github.com/pserwylo/beat-game/actions/runs/599157891

**Comment by pserwylo** [comment]
Okay, pre-release of the latest v0.3.0 version is now available here, let me know how it goes!

https://github.com/pserwylo/beat-game/releases/tag/debug-build-v0.3.0

Obviously any feedback is very welcome (including whether the build even installs for you).

This is a *debug build*, you will need to uninstall the F-Droid version in order to install this. Once F-Droid builds, signs, and releases this version, I'll replace this debug-build with the actual release version.

**Reply by bydariogamer** [reply]
Wow! The 0.3.0 is amazing. Works well and I find super-cool the new menus. The achievements are great too.
The new implementation of the world generator is awesome: now it is possible to play for real! ;)
The bullet-through-paper bug is fixed (thanks!) and the new songs are fantastic.
As only misfeature: the new heart system gets stuck when you die so it doesn't show all the empty hearts if you get  lot of damage suddenly. I mean: if you had 40HP and you die, it will continue showing 2 hearts during the ghost animation. I don't know how you handle the graphics... Maybe you just update the screen in the rectangle that the character (who still needs a name :eyes:) occupies... However, showing the 5 empty hearts should be easy :wink:.
